user-storage-jpa/src/test/java/org/keycloak/quickstart/ArquillianJpaStorageTest.java

```diff
@@ -21,54 +21,64 @@
 import org.junit.Before;
+import org.junit.BeforeClass;
 import org.junit.Test;
 import org.junit.runner.RunWith;
-import org.keycloak.quickstart.page.ConsolePage;
 import org.keycloak.quickstart.storage.user.EjbExampleUserStorageProvider;
 import org.keycloak.quickstart.storage.user.EjbExampleUserStorageProviderFactory;
 import org.keycloak.quickstart.storage.user.UserAdapter;
 import org.keycloak.quickstart.storage.user.UserEntity;
+import org.keycloak.representations.idm.ComponentRepresentation;
+import org.keycloak.representations.idm.UserRepresentation;
+import org.keycloak.test.FluentTestsHelper;
 import org.keycloak.test.page.LoginPage;
-import org.openqa.selenium.By;
-import org.openqa.selenium.support.ui.ExpectedConditions;
-import org.openqa.selenium.support.ui.WebDriverWait;
 import org.openqa.selenium.WebDriver;
+import org.wildfly.extras.creaper.core.ManagementClient;
+import org.wildfly.extras.creaper.core.online.ManagementProtocol;
+import org.wildfly.extras.creaper.core.online.OnlineManagementClient;
+import org.wildfly.extras.creaper.core.online.OnlineOptions;
 
-import java.io.IOException;
-import java.net.MalformedURLException;
-import java.net.URL;
+import javax.ws.rs.core.Response;
 import java.util.concurrent.TimeUnit;
 
 import static java.lang.String.format;
-import org.jboss.arquillian.graphene.Graphene;
 import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertNotNull;
-import static org.junit.Assert.fail;
 
 @RunWith(Arquillian.class)
 public class ArquillianJpaStorageTest {
 
-    private static final String KEYCLOAK_URL = "http://%s:%s/auth%s";
 
-    @Page
-    private LoginPage loginPage;
+    public static final String KEYCLOAK_URL = "http://localhost:8180/auth";
+    public static final int KEYCLOAK_MGMT_PORT = Integer.parseInt(System.getProperty("keycloakManagementPort"));
+    public static final String PROVIDER_TYPE = "org.keycloak.storage.UserStorageProvider";
 
     @Page
-    private ConsolePage consolePage;
+    private LoginPage loginPage;
 
     @Drone
     private WebDriver webDriver;
 
-    @ArquillianResource
-    private URL contextRoot;
+    private static FluentTestsHelper testsHelper;
+
+    private static OnlineManagementClient onlineManagementClient;
 
     @Deployment(testable = false)
-    public static Archive<?> createTestArchive() throws IOException {
+    public static Archive<?> createTestArchive() throws Exception {
+        // set storage
+        onlineManagementClient = ManagementClient.online(OnlineOptions
+                .standalone()
+                .hostAndPort("localhost", KEYCLOAK_MGMT_PORT)
+                .protocol(ManagementProtocol.HTTP_REMOTING)
+                .build());
+        onlineManagementClient.execute("xa-data-source add --name=ExampleXADS --driver-name=h2 --jndi-name=java:jboss/datasources/ExampleXADS --xa-datasource-properties={URL=>jdbc:h2:mem:test} --enabled=true");
+        onlineManagementClient.executeCli("reload");
```

Comment:
Thanks for the review! AFAIK the `@Deployment` annotated methods are called even before `@BeforeClass`. The datasource needs to be created before deploying the storage provider, otherwise the deployment fails.